metis-common/src/main/java/eu/europeana/metis/zoho/ZohoAccessClient.java

```diff
@@ -85,26 +83,26 @@ public ZohoAccessClient(String grantToken) throws ZohoOAuthException {
    * @return the contact corresponding to the email
    * @throws BadContentException if an exception occurred during searching
    */
-  public ZCRMRecord getZcrmRecordContactByEmail(String email) throws BadContentException {
+  public Optional<ZCRMRecord> getZcrmRecordContactByEmail(String email) throws BadContentException {
     final BulkAPIResponse bulkAPIResponseContacts;
     try {
       bulkAPIResponseContacts = zcrmModuleContacts.searchByEmail(email);
     } catch (ZCRMException e) {
       throw new BadContentException("Zoho search by email threw an exception", e);
     }
-    final List<ZCRMRecord> zcrmRecords = (List<ZCRMRecord>) bulkAPIResponseContacts.getData();
-    return zcrmRecords.isEmpty() ? null : zcrmRecords.get(0);
+    final List<ZCRMRecord> zcrmRecords = castItemsToType(bulkAPIResponseContacts.getData(),
+        ZCRMRecord.class);
+    return zcrmRecords.stream().findFirst();
   }
 
-
   /**
    * Get a contact by using an organization email.
    *
    * @param organizationName the organization name to search for
    * @return the organization corresponding to the organization name
    * @throws BadContentException if an exception occurred during searching
    */
-  public ZCRMRecord getZcrmRecordOrganizationByName(String organizationName)
+  public Optional<ZCRMRecord> getZcrmRecordOrganizationByName(String organizationName)
```

Comment:
According to my IDE this method, nor the following two, are in use. Is this true?

Reply:
Yes and no.
True for our code base.
False, it's used in some script that the api team is using..

Reply:
Ah, ok. I didn't know that.